dedalus/core/solvers.py

```diff
@@ -133,8 +139,23 @@ def solve_sparse(self, pencil, N, target, rebuild_coeffs=False, **kw):
         # Solve as sparse general eigenvalue problem
         A = pencil.L_exp
         B = -pencil.M_exp
+        # Solve for the right eigenvectors
         self.eigenvalues, self.eigenvectors = scipy_sparse_eigs(A=A, B=B, N=N, target=target, matsolver=self.matsolver, **kw)
+        if left == True:
+            # Solve for the left eigenvectors
+            #TODO: I think A and B are sparse.csr_matrix objects, in which case csr_matrix.getH might be better than np.conjugate(np.transpose()). Need to test.
+            self.left_eigenvalues, self.left_eigenvectors = scipy_sparse_eigs(A=np.conjugate(np.transpose(A)),
+                                                                              B=np.conjugate(np.transpose(B)),
+                                                                              N=N, target=np.conjugate(target),
+                                                                              matsolver=self.matsolver, **kw)
+            # The following isn't an error, just bad luck: sometimes the sparse solver doesn't hit the same targets for left as it does for right eigenvectors
+            if not np.allclose(self.eigenvalues, np.conjugate(self.left_eigenvalues)):
+                logger.warning("Conjugate of left eigenvalues does not match right eigenvalues.")
+            # In absence of above warning, modified_left_eigenvectors forms a biorthogonal set with the right eigenvectors
+            # (perhaps @ instead of * is better)
+            self.modified_left_eigenvectors = np.transpose(np.conjugate(self.left_eigenvectors.T)*-pencil.M)
```

Comment:
Maybe we want a different sign and/or normalization here to make the modified left eigenvectors orthonormal the right eigenvectors?

Reply:
Latest commit demonstrates a few different options for what to do here (except the normalization part). Next time we meet we can chat about what makes the most sense.